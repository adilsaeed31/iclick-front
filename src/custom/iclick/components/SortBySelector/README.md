SortBySelector
=================

## Usage

```jsx
import SortBySelector from "components/SortBySelector";

```
## Props
The component accepts the following props

### className

###
