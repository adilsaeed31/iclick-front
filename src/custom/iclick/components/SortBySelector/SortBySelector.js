import React, { Component } from "react";
import PropTypes from "prop-types";
import Select from "custom/iclick/components/Select";

const SORT_BY = [
  {
    name: "Newest",
    value: "updatedAt"
  },
  {
    name: "Price: low to high",
    value: "minPrice-asc"
  },
  {
    name: "Price: high to low",
    value: "minPrice-desc"
  }
];

class SortBySelector extends Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    sortBy: PropTypes.string.isRequired
  }

  componentDidMount() {
  }

  state = {
    sortBy: "updatedAt",
    direction: "desc"
  }

  handleChange = (event) => {
    this.setState({
      sortBy: event.target.value,
      direction: event.target.value === "minPrice-asc" ? "asc" : event.target.value === "minPrice-desc" ? "desc" : this.state.direction
    }, () => {
      this.props.onChange(this.getSortBy());
    });
  }

  getSortBy() {
    const { sortBy, direction } = this.state;
    if (sortBy === "minPrice-asc" || sortBy === "minPrice-desc") {
      return sortBy;
    }
    return `${sortBy}-${direction}`;
  }

  handleDirection = (e) => {
    e.preventDefault();
    this.setState({
      direction: this.state.direction === "desc" ? "asc" : "desc"
    }, () => {
      this.props.onChange(this.getSortBy());
    });
  }

  render() {
    const { sortBy, labelValue, classes, selectorName } = this.props;
    return (
      <div className={classes}>
        <label>{labelValue}</label>
        <div className="select-custom">
          <Select
            selectorName = {selectorName}
            value={sortBy}
            options={SORT_BY}
            inputProps={{
              name: "sortBy",
              id: "sort-by"
            }}
            onChange={this.handleChange}
          />
        </div>
        <a
          href="#"
          className={`sorter-btn btn-${this.state.direction}`}
          title="Set Ascending Direction"
          onClick={this.handleDirection}
        >
          <span className="sr-only">Set Ascending Direction</span>
        </a>
      </div>

    );
  }
}

export default SortBySelector;
