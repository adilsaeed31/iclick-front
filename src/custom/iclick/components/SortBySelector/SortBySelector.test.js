import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import SortBySelector from "./SortBySelector";

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <SortBySelector
        sortBy={"updatedAt-desc"}
        onChange={() => true}
      />
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
