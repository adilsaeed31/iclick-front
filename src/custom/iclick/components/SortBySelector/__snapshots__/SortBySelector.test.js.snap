// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiInputBase-root-26 MuiInput-root-13 SkSelect-input-5"
  onClick={[Function]}
>
  <div
    className="MuiSelect-root-6"
  >
    <div
      aria-haspopup="true"
      aria-owns={null}
      aria-pressed="false"
      className="MuiSelect-select-7 MuiSelect-selectMenu-10 SkSelect-selectMenu-3 MuiInputBase-input-36 MuiInput-input-21"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Newest
    </div>
    <input
      id="sort-by"
      name="sortBy"
      type="hidden"
      value="updatedAt-desc"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-43 MuiSelect-icon-12"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
